app/api/ada/adaTransactions/adaNewTransactions.js

```diff
@@ -33,15 +33,25 @@ export async function getAdaTransactionFee(
   amount: string,
   changeAdaAddr: ?AdaAddress,
   possibleInputAddresses: Array<AdaAddress>,
-  getUTXOsForAddresses: AddressUtxoFunc
+  getUTXOsForAddresses: AddressUtxoFunc,
+  shouldSendAll: boolean
 ): Promise<AdaFeeEstimateResponse> {
-  const { txBuilder } = await newAdaUnsignedTx(
-    receiver,
-    amount,
-    changeAdaAddr,
-    possibleInputAddresses,
-    getUTXOsForAddresses
-  );
+  let txBuilder;
+  if (shouldSendAll) {
+    txBuilder = await sendAllUnsignedTx(
```

Comment:
I can't understand the issue considering that I changed something like

const { bar} =  await foo()

to

let bar;
bar = await foo()


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ app/api/ada/adaTransactions/adaNewTransactions.js:41:23

Cannot call await with sendAllUnsignedTx(...) bound to p because:
 • Either cannot call txBuilder.get_balance_without_fees because:
    • Either property get_balance_without_fees is missing in UnsignedTxFromUtxoResponse [1].
    • Or property get_balance_without_fees is missing in object type [2].
 • Or cannot call txBuilder.get_balance_without_fees because property get_balance_without_fees is missing in
   Promise [3].

        app/api/ada/adaTransactions/adaNewTransactions.js
         38│ ): Promise<AdaFeeEstimateResponse> {
         39│   let txBuilder;
         40│   if (shouldSendAll) {
         41│     txBuilder = await sendAllUnsignedTx(
         42│       receiver,
         43│       possibleInputAddresses,
         44│       getUTXOsForAddresses
         45│     );
         46│   } else {
         47│     txBuilder = await newAdaUnsignedTx(
         48│       receiver,
           :
    [3]  72│ ): Promise<UnsignedTxResponse> {

        app/api/ada/adaTypes.js
 [1][2] 133│ export type UnsignedTxResponse = UnsignedTxFromUtxoResponse & {
        134│   addressesMap: AdaAddressMap,
        135│ };


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ app/api/ada/adaTransactions/adaNewTransactions.js:47:23

Cannot call await with newAdaUnsignedTx(...) bound to p because:
 • Either cannot call txBuilder.get_balance_without_fees because:
    • Either property get_balance_without_fees is missing in UnsignedTxFromUtxoResponse [1].
    • Or property get_balance_without_fees is missing in object type [2].
 • Or cannot call txBuilder.get_balance_without_fees because property get_balance_without_fees is missing in
   Promise [3].

        app/api/ada/adaTransactions/adaNewTransactions.js
         44│       getUTXOsForAddresses
         45│     );
         46│   } else {
         47│     txBuilder = await newAdaUnsignedTx(
         48│       receiver,
         49│       amount,
         50│       changeAdaAddr,
         51│       possibleInputAddresses,
         52│       getUTXOsForAddresses
         53│     );
         54│   }
         55│   /**
         56│    * Note: get_balance_without_fees() != estimated fee
           :
    [3] 142│ ): Promise<UnsignedTxResponse> {

        app/api/ada/adaTypes.js
 [1][2] 133│ export type UnsignedTxResponse = UnsignedTxFromUtxoResponse & {
        134│   addressesMap: AdaAddressMap,
        135│ };



Found 2 errors

Reply:
```const { bar} = await foo()```

foo() returns an object (dictionary / hashmap) and then you subtract the content for the bar key.

```
let bar;
bar = await foo()
```
this is different because you assigned the result object from foo() to bar